tensorflow/contrib/go/tensor.go

```diff
@@ -0,0 +1,763 @@
+package tensorflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/tensorflow/tensorflow/tensorflow/contrib/go/proto"
+)
+
+import "C"
+
+const (
+	cAckByte  = 6
+	cBellByte = 7
+	cDc1      = 17
+
+	cBytesComplex64 = 8
+	cBytesFloat32   = 4
+	cBytesFloat64   = 8
+	cBytesInt16     = 2
+	cBytesInt32     = 4
+	cBytesInt64     = 8
+	cBytesUint16    = 2
+)
+
+// DataType Type of the data contained by a Tensor
+type DataType pb.DataType
+
+// TensorInt Interface to be implemented by the tensors.
+type TensorInt interface {
```

Comment:
TensorInt is slightly confusing naming to me - it could read easily as "Tensor Integer".  Is there a more unique name it could have, such as TensorIface?  Or, following the standard Go conventions, some kind of agent noun ending on -er (which is unfortunate with Tensor, because Tensorer is horrible! :).  Tensorable is awkward but might do.  Or just TensorInterface - it's not that long to type.